packages/core/src/getNextSnapshot.ts

```diff
@@ -0,0 +1,74 @@
+import { createEmptyActor } from './actors/index.ts';
+import {
+  ActorRefFrom,
+  ActorScope,
+  AnyActorLogic,
+  AnyActorScope,
+  EventFromLogic,
+  InputFrom,
+  SnapshotFrom
+} from './types.ts';
+
+/** @internal */
+export function createInertActorScope<T extends AnyActorLogic>(
+  _actorLogic: T
+): AnyActorScope {
+  const self = createEmptyActor() as ActorRefFrom<T>;
```

Comment:
Sample case why this is unsafe:
```ts
createMachine({
  entry: assign(({ self }) => {
     self.getSnapshot().value // oops! it throws
     return {}
  })
})
```

`createEmptyActor` returns `Snapshot<undefined>` so it's not compatible with the snapshot type of the actual `self` that an actor (and its `transition`) expects

Reply:
We can change this with `const self = createActor(actorLogic)`

Reply:
This would definitely be better 👍